src/workflows/airqo_etl_utils/data_api.py

```
@@ -329,6 +328,33 @@ def get_devices_by_network(
+            - The method first checks for maintenance activities. If none are found, it falls back to deployment activities.
+            - If no activities are available, the method returns None.
+        """
+        try:
+            activities = device.get("latest_activities_by_type", {}).get("maintenance")
+            if not activities:
+                activities = device.get("latest_activities_by_type", {}).get(
+                    "deployment"
+                )
+            return activities.get("date") if activities else None
+        except Exception as e:
+            logger.exception(f"Failed to fetch maintenance activities: {e}")
```

Ruff (0.12.2)

Redundant exception object included in `logging.exception` call

(TRY401)